### ISSUE TITLE
ClassCastException handling in JavaFxUtils.fr lookupNode doesn't seem to be correct

### ISSUE THREAD

**Issue by alaendle** [Frege/FregeFX#9]
Unfortunately I'm a newbie to frege - but at least I'm relatively convinced that the code in JavaFxUtils.fr tries to hide ClassCastExceptions if the found Node couldn't be casted. 

To give an example - just change the Hello World example the following way.

```
             button `action_` (button.setText "Thanks!")
         vbox <: do
             button <- Button.new "Another one"
-            button `action_` (button.setText "Cool!")
+            button `action_` do
+              scene <- stage.getScene
+              ta <- (lookupNode scene "*" :: IO TextArea)
+              button.setText "Cool!"
         -- experiment that dynamically adds 5 buttons
         vbox `addAll` map (Button.new . ("Button "++) . show) [1..5]
```

If you now run with gradlew run and click the Another one" button an ClassCastException gets thrown.

Unfortunately I didn't know a better solution (otherwise I would have sent an pull-request).
Please excuse if I didn't interpreted the code correctly or if this isn't an issue.

Best regards,
Andreas

**Comment by Dierk** [comment]
Hi Andreas,

you make a valid point. The Java CCE runs through and is not handled by `lookupNode` as it should be (it appears that `either` is not called as I think it should. I have to investigate).
However, you would get a runtime error either way (and the Java exception gives an arguably better indication of what is wrong).

`lookupNode` is the _unsafe, best effort_ convenience function around `lookup`. It's the function to use if you are either dead-sure that your node exists and will be found and if you are ok with failing with an exception otherwise. You can at any point just use the safe `lookup` and deal with lookup and cast errors yourself.

I would be very happy about any suggestion for improvement.

Thanks for posting and for using FregeFX!

**Comment by alaendle** [comment]
Thanks for your explanation - inspired by the implementation of frege swing I guess I've found a way to change the code to our expectations.

I removed the Caster class and replaced it's usages with simple, direct cast statements.

Patch for JavaFxUtils.fr (starting at line 107):

```
         Just node -> downcast node >>= either (\cce -> error $ "cannot cast selector '" ++ selector ++ "'") return
         Nothing -> error $ "cannot find '" ++ selector ++ "'"

--- native Java implementation of the downcast since we cannot do that in Frege.
-native module where {
-    public static class Caster {
-        public static <T> T cast(T t) {
-            return t;
-        }
-    }
-}
-
 data ClassCastException = native java.lang.ClassCastException

 -- specific downcasts
-native toTextArea "Caster.cast" :: Node -> IO (ClassCastException | TextArea    )
-native toListView "Caster.cast" :: Node -> IO (ClassCastException | ListView t  )
+native toTextArea "(javafx.scene.control.TextArea)" :: Node -> IO (ClassCastException | TextArea    )
+native toListView "(javafx.scene.control.ListView)" :: Node -> IO (ClassCastException | ListView t  )


 -- generalized downcast
```

**Comment by Dierk** [comment]
cool!

fixed with #fdfb734
in 0.2.1-SNAPSHOT

thanks!
